.github/workflows/create_sbom_report.yml

```diff
@@ -0,0 +1,88 @@
+name: Create and upload a SBOM to release assets
+# Inherited variables:
+# github.event.client_payload.imageLabel - AzDO image label
+# github.event.client_payload.GHreleaseID - Current release ID
+# github.event.client_payload.imageVersion - AzDO image version "major.minor"
+# github.event.client_payload.ReleaseBranchName - Bacause it wouldn't work withoit it, sorry
+# 
+# Current SYFT tool issues:
+# macOS (minor): very long cataloging process (more than 6 hours) (https://github.com/anchore/syft/issues/1328), 
+# macOS (major): promt privilegies that blocking process indefinetely (https://github.com/anchore/syft/issues/1367)
+on:
+  repository_dispatch:
+    types: [generate-sbom]
+defaults:
+  run:
+    shell: pwsh
+jobs:
+  #Converting image OS variable for the next steps
+  initialize: 
+    runs-on: ubuntu-latest
+    outputs:
+      agent-spec: ${{ steps.converter.outputs.current-os }}
+    steps:
+      - name: Convert image label variable for ${{ github.event.client_payload.ReleaseBranchName }}
+        id: converter
+        run: |
+          $imageLabel = "${{ github.event.client_payload.imageLabel }}"
+          $currentOS = switch ($imageLabel) {
+            'ubuntu22' { "ubuntu-22.04" }
+            'ubuntu20' { "ubuntu-20.04" }
+            'ubuntu18' { "ubuntu-18.04" }
+            'win22' { "windows-2022" }
+            'win19' { "windows-2019" }
+            'macOS-12' { "macos-12" }
+            'macOS-11' { "macos-11" }
+            default {
+              echo "currentOS variable is undefined. Please check imageLabel."
+              exit 1  
+            }
+          }
+          "current-os=$currentOS" | Out-File -FilePath $env:GITHUB_OUTPUT -Append
+  #Checking image version on available runner
+  version-check:
+    needs: initialize
+    runs-on: ${{ needs.initialize.output.agent-spec }}
+    steps:
+      - name: Available image version check
+        run: |
+          if ($env:ImageVersion -ne ${{ github.event.client_payload.imageVersion }}) {
```

Comment:
Use single quotes around payload variables: `if ($env:ImageVersion -ne '${{ github.event.client_payload.imageVersion }}') {`.
Otherwise, GH will compare string with double.

Reply:
Accepted and resolved